sdk/core/azure-core-test/src/test/java/com/azure/core/test/TestProxyTests.java

```diff
@@ -144,6 +148,25 @@ public void testMismatch() {
         assertTrue(thrown.getMessage().contains("Uri doesn't match"));
     }
 
+    @Test
+    @Tag("Record")
+    public void testResetTestProxyData() throws MalformedURLException {
+        HttpURLConnectionHttpClient client = new HttpURLConnectionHttpClient();
+        HttpPipelinePolicy auditorPolicy = (context, next) -> {
+            HttpHeaders headers = context.getHttpRequest().getHeaders();
+            String headerValue = headers.getValue(HttpHeaderName.fromString("x-recording-upstream-base-uri"));
+            Assertions.assertNull(headerValue);
+            return next.process();
```

Comment:
Does this need to check the return here? My impression is that this policy as written will check the outgoing request before it is modified by `TestProxyRecordingPolicy`.